### DISCUSSION TITLE
Disallow date fields from working with Date objects?

### DISCUSSION THREAD

**Discussion by IlyaSemenov** [romeerez/orchid-orm#134] [General]
I noticed in raw SQL logs that when I insert a date column and pass an ISO date, it's converted to Javascript Date before it's sent to the database:

```ts
class UserTable extends BaseTable {
  readonly table = "user"

  columns = this.setColumns((t) => ({
    id: t.identity().primaryKey(),
    dob: t.date(),
  }))
}

await db.user.create({ dob: "2000-10-20" })
// INSERT INTO "user"("dob") VALUES ($1) RETURNING * ['2000-10-20T00:00:00.000Z']
// How so? I never passed and never wanted any time values.
```

I then noticed that I can even pass a raw Date object:

```ts
await db.user.create({ dob: new Date() })
// (3.1ms) INSERT INTO "user"("dob") VALUES ($1) RETURNING * ['2023-06-26T09:38:50.049Z']
```

I think this is a time bomb for timezone errors and should be disallowed. In the `new Date()` example above, the actual value saved to the database will depend on the time zone of the Postgres host. If the app works in EST and the database runs on a server in UTC, I think `new Date()` will save the tomorrow's date when it's 9pm in New York - something that a naive user wouldn't expect. Moreover, this kind of problems could be left unnoticed in unit tests when the server and the app work on the same server.

So my point here is that `t.date()` should better work with strings only to prevent misuse (possibly allowing to opt-in for dates with `asDate` for those who are absolutely sure they can handle tz offsets properly). What do you think?

**Comment by romeerez** [comment]
I saved it to an issue to work on it later: https://github.com/romeerez/orchid-orm/issues/179
